### PR TITLE
Core/Opcodes: SMSG_INIT_WORLD_STATES

### PR DESCRIPTION
**Pull request by Golrag**
WorldPackets::WorldState::InitWorldStates is now used :)

### CODE REVIEW

**Comment by DDuarte** [comment]
:+1: :sparkles:

File: src/server/game/Battlefield/Zones/BattlefieldWG.cpp [hunk 1]
```diff
@@ -1550,9 +1549,9 @@ void WintergraspWorkshop::UpdateGraveyardAndWorkshop()
         GiveControlTo(_wg->GetDefenderTeam(), true);
 }
 
-void WintergraspWorkshop::FillInitialWorldStates(WorldPacket& data)
+void WintergraspWorkshop::FillInitialWorldStates(WorldPackets::WorldState::InitWorldStates& packet)
 {
-    data << uint32(_staticInfo->WorldStateId) << uint32(_state);
+    packet.Worldstates.push_back({ uint32(_staticInfo->WorldStateId), int32(_state) });
```
Comment (joschiwald) [inline src/server/game/Battlefield/Zones/BattlefieldWG.cpp]: just a hint:
```c++
packet.Worldstates.emplace_back(_staticInfo->WorldStateId, _state);
```
looks more clean

**Comment by joschiwald** [comment]
long time ago i tried to get rid of these tons of worldstate inits by caching them in map class, stopped working on them for some reason, can't remember

File: src/server/game/Battlefield/Zones/BattlefieldWG.cpp (hunk 1 above)
Comment (Golrag) [inline src/server/game/Battlefield/Zones/BattlefieldWG.cpp]: So ehm, change every push_back to emplace_back ? ._.

Comment (DDuarte) [inline src/server/game/Battlefield/Zones/BattlefieldWG.cpp]: Yes but first verify that it compiles with a single emplace_back.
If you need help doing a regex to change all of them do say, please don't do it manually :P

**Comment by Golrag** [comment]
Updated ! :)

File: src/server/game/Entities/Player/Player.cpp [hunk 2]
```diff
@@ -8923,30 +8923,26 @@ void Player::SendInitWorldStates(uint32 zoneid, uint32 areaid)
...
+    packet.Worldstates.emplace_back(2264, 0);              // 1
+    packet.Worldstates.emplace_back(2263, 0);              // 2
+    packet.Worldstates.emplace_back(2262, 0);              // 3
+    packet.Worldstates.emplace_back(2261, 0);              // 4
+    packet.Worldstates.emplace_back(2260, 0);              // 5
+    packet.Worldstates.emplace_back(2259, 0);              // 6
+
+    packet.Worldstates.emplace_back(3191, int32(sWorld->getBoolConfig(CONFIG_ARENA_SEASON_IN_PROGRESS) ? sWorld->getIntConfig(CONFIG_ARENA_SEASON_ID) : 0)); // 7 Current Season - Arena season in progress
+                                                                                                                                                              // 0 - End of season
+    packet.Worldstates.emplace_back(3901, int32(sWorld->getIntConfig(CONFIG_ARENA_SEASON_ID) - sWorld->getBoolConfig(CONFIG_ARENA_SEASON_IN_PROGRESS)));     // 8 PreviousSeason
```
Comment (DDuarte) [inline src/server/game/Entities/Player/Player.cpp]: This is slightly different, what happened?

**Pull request merged**